### PR TITLE
py-elasticsearch: new versions

### PR DESCRIPTION
**Pull request by heerener**
Also add py-elastic-transport as a new dependency

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-elasticsearch/package.py [hunk 1]
```diff
@@ -12,13 +12,22 @@ class PyElasticsearch(PythonPackage):
...
     version("5.2.0", sha256="45d9f8fbe0878a1b7493afeb20f4f6677a43982776ed1a77d9373e9c5b9de966")
     version("2.3.0", sha256="be3080a2bf32dff0a9f9fcc1c087515a25a357645673a976d25ef77166134d81")
 
+    variant("async", when="@8.6.2:", default=False, description="Include support for asyncio")
+
     depends_on("py-setuptools", type="build")
-    depends_on("py-urllib3@1.8:1", type=("build", "run"))
+    depends_on("python@3.7:3", when="@7.6.0:", type=("build", "run"))
+    depends_on("py-urllib3@1.8:1", when="@:5.2.0", type=("build", "run"))
+    depends_on("py-urllib3@1.21.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elasticsearch/package.py]: urllib3 doesn't seem to be a dep in the latest version

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by tldahlgren — COMMENTED** [review]
Confirmed the three version sha256.

File: var/spack/repos/builtin/packages/py-elasticsearch/package.py [hunk 2]
```diff
@@ -12,13 +12,22 @@ class PyElasticsearch(PythonPackage):
...
     version("5.2.0", sha256="45d9f8fbe0878a1b7493afeb20f4f6677a43982776ed1a77d9373e9c5b9de966")
     version("2.3.0", sha256="be3080a2bf32dff0a9f9fcc1c087515a25a357645673a976d25ef77166134d81")
 
+    variant("async", when="@8.6.2:", default=False, description="Include support for asyncio")
+
     depends_on("py-setuptools", type="build")
-    depends_on("py-urllib3@1.8:1", type=("build", "run"))
+    depends_on("python@3.7:3", when="@7.6.0:", type=("build", "run"))
+    depends_on("py-urllib3@1.8:1", when="@:5.2.0", type=("build", "run"))
+    depends_on("py-urllib3@1.21.1:7.6.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elasticsearch/package.py]: I think you meant?
```suggestion
    depends_on("py-urllib3@1.21.1:", when="@6:7", type=("build", "run"))
```

Comment (heerener) [reply var/spack/repos/builtin/packages/py-elasticsearch/package.py]: I did and can't even claim "don't commit and Friday afternoon" in my defense...

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**